addons/account_bank_statement_import_qif/account_bank_statement_import_qif.py

```diff
@@ -3,42 +3,34 @@
 import dateutil.parser
 import StringIO
 
-from openerp.tools.translate import _
-from openerp.osv import osv, fields
+from openerp import api, fields, models, _
 from openerp.exceptions import UserError
 
-class account_bank_statement_import(osv.TransientModel):
-    _inherit = "account.bank.statement.import"
 
-    _columns = {
-        'journal_id': fields.many2one('account.journal', string='Journal', help='Accounting journal related to the bank statement you\'re importing. It has be be manually chosen for statement formats which doesn\'t allow automatic journal detection (QIF for example).'),
-        'hide_journal_field': fields.boolean('Hide the journal field in the view'),
-    }
+class account_bank_statement_import(models.TransientModel):
+    _inherit = "account.bank.statement.import"
 
-    def _get_hide_journal_field(self, cr, uid, context=None):
-        return context and 'journal_id' in context or False
+    def _get_hide_journal_field(self):
+        return self._context and 'journal_id' in self._context or False
 
-    _defaults = {
-        'hide_journal_field': _get_hide_journal_field,
-    }
+    journal_id = fields.Many2one('account.journal', string='Journal', help='Accounting journal related to the bank statement you\'re importing. It has be be manually chosen for statement formats which doesn\'t allow automatic journal detection (QIF for example).')
+    hide_journal_field = fields.Boolean(string='Hide the journal field in the view', default=_get_hide_journal_field)
 
-    def _get_journal(self, cr, uid, currency_id, bank_account_id, account_number, context=None):
+    def _get_journal(self, currency_id, bank_account_id, account_number):
         """ As .QIF format does not allow us to detect the journal, we need to let the user choose it.
             We set it in context before to call super so it's the same as calling the widget from a journal """
-        if context is None:
-            context = {}
-        if context.get('active_id'):
-            record = self.browse(cr, uid, context.get('active_id'), context=context)
+        if self._context.get('active_id'):
+            record = self.browse(self._context.get('active_id'))
             if record.journal_id:
-                context['journal_id'] = record.journal_id.id
-        return super(account_bank_statement_import, self)._get_journal(cr, uid, currency_id, bank_account_id, account_number, context=context)
+                self = self.with_context(journal_id=record.journal_id.id)
```

Comment:
Reassigning self in a method is probably unnecessary and may break translation introspection !!!